src/composables/airGap.ts

```diff
@@ -0,0 +1,157 @@
+
+      const dataUint8Array = bs58check.decode(serializedData!);
+      const ur = UR.fromBuffer(Buffer.from(dataUint8Array));
+
+      // Set the chunk sizes for single-chunk and multi-chunk encoding.
+      const SETTINGS_SERIALIZER_SINGLE_CHUNK_SIZE = 500;
+      const SETTINGS_SERIALIZER_MULTI_CHUNK_SIZE = 250;
+
+      // Create a UR encoder for single-chunk encoding.
+      const singleEncoder = new UREncoder(
+        ur,
+        SETTINGS_SERIALIZER_SINGLE_CHUNK_SIZE,
+      );
+      // If the UR requires multi-chunk encoding,
+      // create a new UR encoder with the appropriate chunk size.
+      if (singleEncoder.fragmentsLength !== 1) {
+        const multiEncoder = new UREncoder(
+          ur,
+          SETTINGS_SERIALIZER_MULTI_CHUNK_SIZE,
+        );
+        const fragments: string[] = [];
+
+        [...Array(multiEncoder.fragmentsLength)].forEach(() => {
+          fragments.push(multiEncoder.nextPart());
+        });
+
+        return fragments;
+      }
+
+      // Encode the UR and return the UR string in upper case.
+      return [(singleEncoder.nextPart()).toUpperCase()];
+    } catch (error) {
+      handleUnknownError(error);
+      return [];
+    }
+  }
+
+  async function deserializeData(data: string) {
+    await watchUntilTruthy(serializer);
+    return serializer.value?.deserialize(data);
+  }
+
+  /**
+   * Extracts shared accounts from deserialized data.
+   */
+  async function extractAccountShareResponseData(
+    data: IACMessageDefinitionObjectV3[] = [],
+  ): Promise<IAccountRaw[]> {
+    return Promise.all(
+      data
+        .filter((item) => item.type === IACMessageType.AccountShareResponse)
+        .map(async (item) => {
+          const aeProtocol = new AeternityProtocol();
+          const address = await aeProtocol.getAddressFromPublicKey(
+            (item.payload as AccountShareResponse).publicKey,
+          ) as Encoded.AccountAddress;
+
+          return {
+            address,
+            type: ACCOUNT_AIR_GAP_WALLET,
+            airGapPublicKey: (item.payload as AccountShareResponse).publicKey,
+            protocol: PROTOCOLS.aeternity,
+            isRestored: false,
+          };
+        }),
+    );
+  }
+
+  /**
+   * Extracts the signed transaction response data from deserialized data
+   * and returns the transaction object.
+   */
+  async function extractSignedTransactionResponseData(
+    data: IACMessageDefinitionObjectV3[] = [],
+  ): Promise<string | null> {
+    const payload = data.find(
+      (item) => item.type === IACMessageType.TransactionSignResponse,
+    )?.payload as any;
+
+    return payload?.transaction || null;
+  }
+
+  async function generateTransactionURDataFragments(
+    publicKey: string,
+    transaction: string,
+    networkId: string,
+  ): Promise<string[]> {
+    const id = Math.floor(Math.random() * 90000000 + 10000000);
+    const callbackURL = `${MOBILE_SCHEMA}?d=`;
+    const payload: any = {
```

Comment:
To avoid using `any` I suggest to assign the value directly in the `messageDefinitionObject` definition block. This way the `IACMessageDefinitionObjectV3` type would nicely validate the schema.

Reply:
It seems `any` was not needed obsolete there